src/main/java/frc/lib/io/vision/VisionIOPhotonVision.java

```diff
@@ -15,102 +15,151 @@
-import edu.wpi.first.units.measure.Distance;
-import frc.lib.util.Timestamped;
-import static edu.wpi.first.units.Units.Meters;
+import edu.wpi.first.math.numbers.N1;
+import edu.wpi.first.math.numbers.N3;
+import edu.wpi.first.math.numbers.N8;
+import edu.wpi.first.units.measure.Time;
+import edu.wpi.first.wpilibj.Alert;
+import edu.wpi.first.wpilibj.Alert.AlertType;
+import static edu.wpi.first.units.Units.Degrees;
 import static edu.wpi.first.units.Units.Seconds;
 import java.util.ArrayList;
-import java.util.HashSet;
 import java.util.List;
 import java.util.Optional;
-import java.util.Set;
-import org.apache.commons.lang3.ArrayUtils;
-import org.photonvision.EstimatedRobotPose;
 import org.photonvision.PhotonCamera;
-import org.photonvision.PhotonPoseEstimator;
-import org.photonvision.PhotonPoseEstimator.PoseStrategy;
 import org.photonvision.targeting.PhotonPipelineResult;
-import org.photonvision.targeting.PhotonTrackedTarget;
 
 /** IO implementation for real PhotonVision hardware. */
 public class VisionIOPhotonVision implements VisionIO {
+    private static final Matrix<N3, N3> DEFAULT_CAMERA_INSTRINSICS =
+        MatBuilder.fill(Nat.N3(), Nat.N3(),
+            2002.948392331919, 0.0, 783.9099067246102,
+            0.0, 1999.0390684862123, 662.7694019679813,
+            0.0, 0.0, 1.0);
+
+    private static final Vector<N8> DEFAULT_DIST_COEFFS = VecBuilder.fill(
+        0.09905119793103302,
+        -0.06388083628565337,
+        3.87402720846368E-5,
+        1.4421218015997156E-4,
+        -0.16329892957216433,
+        -0.004599206903333014,
+        0.0029050841273878885,
+        0.0067195798658376375);
+
     protected final PhotonCamera camera;
-    protected final PhotonPoseEstimator poseEstimator;
+    private final Camera cameraProperties;
+
+    private final Alert failedToFetchInstrinsicsAlert;
 
     /**
      * Creates a new VisionIOPhotonVision.
      *
-     * @param name The configured name of the camera.
-     * @param robotToCamera The 3D position of the camera relative to the robot.
+     * @param name The name of the camera,
+     * @param robotToCamera The transform from the robot to the camera
      */
-    public VisionIOPhotonVision(String name, Transform3d robotToCamera,
-        AprilTagFieldLayout fieldLayout, PoseStrategy strategy)
+    public VisionIOPhotonVision(String name, Transform3d robotToCamera)
     {
         camera = new PhotonCamera(name);
-        poseEstimator = new PhotonPoseEstimator(fieldLayout, strategy, robotToCamera);
-    }
+        failedToFetchInstrinsicsAlert = new Alert(
+            "FAILED TO FETCH INTRINSICS FOR CAMERA " + name
+                + ". CONTINUING WITH DEFAULT (THRIFTYCAM)",
+            AlertType.kError);
 
-    @Override
-    public void updateInputs(VisionIOInputs inputs, Timestamped<Rotation2d> timestampedHeading)
-    {
-        inputs.connected = camera.isConnected();
+        Matrix<N3, N3> cameraIntrinsics = camera.getCameraMatrix().orElseGet(() -> {
```

Comment:
If the co-processors aren't up and running, it likely won't be able to pull this info on boot. Maybe it would be better to pull these from the camera config json and then check that they match what the coproc reports?